### PR TITLE
feat(Template): Add manifest.json to WASM template

### LINKED ISSUE #1426: Add manifest.json in the default Uno WASM app solution template
## What would you like to be added:

1. https://twitter.com/mzikmunddev/status/1165163264398901248?s=12
1. https://nicksnettravels.builttoroam.com/uno-as-a-pwa/

## Why is this needed:

image

## For which Platform:

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] Windows
- [x] Solution Templates

### PR DESCRIPTION
**Pull request by trungnt2910**
This allows the default templates for WASM to generate a PWA that can be installed to the local operating system.

Default icons are automatically included.

GitHub Issue (If applicable): closes #1426

## PR Type

What kind of change does this PR introduce?

- Other... Please describe: Project template changes.

## What is the current behavior?

The Uno Platform project templates for UWP/Xamarin, UWP/.NET 6 and WinUI creates normal web applications for WASM.

## What is the new behavior?

The Uno Platform project templates create PWAs for WASM with a manifest file by default.

image

image

image

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Review by MartinZikmund — APPROVED** [review]

**Comment by MartinZikmund** [comment]
Very nice! Let's wait for @jeromelaban to confirm - I remember he had some concerns regarding caching of PWAs

File: src/SolutionTemplate/UnoSolutionTemplate.WinUI.net6/Wasm/manifest.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{
```
Comment (jeromelaban) [inline src/SolutionTemplate/UnoSolutionTemplate.WinUI.net6/Wasm/manifest.json:1]: Adding a manifest file tends to change the way the have behaves with regards to caching. I think we should be adding an option in the new project experience, as well as in the `dotnet new`.

File: src/SolutionTemplate/UnoSolutionTemplate/Wasm/Wasm.vstemplate [hunk 2]
```diff
@@ -20,7 +20,13 @@
 	<Project TargetFileName="$ext_safeprojectname$.Wasm.csproj" File="UnoQuickStart.Wasm.csproj" ReplaceParameters="true">
 	  <ProjectItem ReplaceParameters="true" TargetFileName="Program.cs">Program.cs</ProjectItem>
 	  <ProjectItem ReplaceParameters="true">LinkerConfig.xml</ProjectItem>
+	  <ProjectItem ReplaceParameters="true" TargetFileName="manifest.json">manifest.json</ProjectItem>
```
Comment (jeromelaban) [inline src/SolutionTemplate/UnoSolutionTemplate/Wasm/Wasm.vstemplate:23]: Let's only add the support for PWA in the net6 default template. The UWP template is less used now, and it does not have a configuration dialogs. Developers can add it manually afterwards.

File: src/SolutionTemplate/UnoSolutionTemplate.WinUI.net6/Wasm/manifest.json (hunk 1 above)
Comment (jeromelaban) [reply src/SolutionTemplate/UnoSolutionTemplate.WinUI.net6/Wasm/manifest.json:1]: See [here](https://platform.uno/docs/articles/uno-development/debugging-templates.html).

Comment (trungnt2910) [reply src/SolutionTemplate/UnoSolutionTemplate.WinUI.net6/Wasm/manifest.json:1]: Addressed this in the next commit.

**Comment by trungnt2910** [comment]
```
  MSBUILD : error : Generation failed: System.AggregateException: One or more errors occurred. ---> System.InvalidOperationException: Generation failed for Uno.UI.SourceGenerators.XamlGenerator.XamlCodeGenerator. System.AggregateException: One or more errors occurred. ---> System.Exception: Processing failed for file D:\a\1\s\UnoAppAll\UnoAppAll.Shared\MainPage.xaml (System.InvalidOperationException: Unable to resolve AppKit.NSView (Reason: None, Location:Unknown, Candidates: ) [D:\a\1\s\UnoAppAll\UnoAppAll.macOS\UnoAppAll.macOS.csproj]
```

```
 Unable to resolve AppKit.NSView
```

This should not be related to my PR.

**Review by carldebilly — APPROVED** [review]

**Comment by jeromelaban** [comment]
It is not... It's a Xamarin issue: https://developercommunity.visualstudio.com/t/post/10164443. We're still assessing how we can work around it.

**Comment by MartinZikmund** [comment]
@trungnt2910 If you resolve the merge conflicts, the PR should pass CI now

**Pull request merged**